### ISSUE TITLE
fatal error in server.go

### ISSUE THREAD

**Issue by halseth** [lightningnetwork/lnd#352]
During a startup/sync of `lnd` a concurrent map access happened in `server.go`, causing `lnd` to crash:
```
fatal error: concurrent map read and map write

goroutine 49210 [running]:
runtime.throw(0x1982028, 0x21)
	/usr/local/Cellar/go/1.9/libexec/src/runtime/panic.go:605 +0x95 fp=0xc422fc4e40 sp=0xc422fc4e20 pc=0x102d4f5
runtime.mapaccess2_faststr(0x184e5e0, 0xc420236210, 0xc4208f4420, 0x21, 0x1, 0x1)
	/usr/local/Cellar/go/1.9/libexec/src/runtime/hashmap_fast.go:324 +0x47a fp=0xc422fc4e98 sp=0xc422fc4e40 pc=0x100e12a
main.(*server).peerTerminationWatcher(0xc420272000, 0xc420c23e40)
	/Users/johan/golang/src/github.com/lightningnetwork/lnd/server.go:1008 +0x597 fp=0xc422fc4fd0 sp=0xc422fc4e98 pc=0x1745207
runtime.goexit()
	/usr/local/Cellar/go/1.9/libexec/src/runtime/asm_amd64.s:2337 +0x1 fp=0xc422fc4fd8 sp=0xc422fc4fd0 pc=0x105daf1
created by main.(*server).addPeer
	/Users/johan/golang/src/github.com/lightningnetwork/lnd/server.go:1252 +0x1a6
```

The map in question is `persistentConnReqs`. Attached are complete crash logs. Happened on commit `d34e131426d378cd58c85d1418432a025eaba867`.

[server_crash.txt](https://github.com/lightningnetwork/lnd/files/1332621/server_crash.txt)

**Comment by halseth** [comment]
Just saw #343. Yo @cfromknecht is it addressing this issue?

**Comment by halseth** [comment]
Did experience the same crash on the `s.peersByPub` map, on commit `80b703b941b8278e37e84933d902b357008cc0c8`:

```
fatal error: concurrent map read and map write

goroutine 14517 [running]:
runtime.throw(0x19831e5, 0x21)
	/usr/local/Cellar/go/1.9/libexec/src/runtime/panic.go:605 +0x95 fp=0xc422678e08 sp=0xc422678de8 pc=0x102d4f5
runtime.mapaccess2_faststr(0x184f160, 0xc4201503f0, 0xc420942870, 0x21, 0xc4202d0088, 0x1)
	/usr/local/Cellar/go/1.9/libexec/src/runtime/hashmap_fast.go:324 +0x47a fp=0xc422678e60 sp=0xc422678e08 pc=0x100e12a
main.(*server).OutboundPeerConnected(0xc4202d4000, 0xc421e80230, 0x1f5ea00, 0xc421861680)
	/Users/johan/golang/src/github.com/lightningnetwork/lnd/server.go:1204 +0x417 fp=0xc422678f98 sp=0xc422678e60 pc=0x1746a87
main.(*server).OutboundPeerConnected-fm(0xc421e80230, 0x1f5ea00, 0xc421861680)
	/Users/johan/golang/src/github.com/lightningnetwork/lnd/server.go:324 +0x48 fp=0xc422678fc8 sp=0xc422678f98 pc=0x175a258
runtime.goexit()
	/usr/local/Cellar/go/1.9/libexec/src/runtime/asm_amd64.s:2337 +0x1 fp=0xc422678fd0 sp=0xc422678fc8 pc=0x105daf1
created by github.com/lightningnetwork/lnd/vendor/github.com/roasbeef/btcd/connmgr.(*ConnManager).connHandler
	/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/roasbeef/btcd/connmgr/connmanager.go:236 +0x3db
```
[server_crash2.txt](https://github.com/lightningnetwork/lnd/files/1336766/server_crash2.txt)

**Comment by cfromknecht** [comment]
I believe this is the same bug! Previous design only properly handled the case where the server initiated the disconnect, but could cause this error if the remote peer disconnects unexpectedly. New design should prevent the unprotected access to the internal state

**Comment by cfromknecht** [comment]
I believe this issue can be closed now, fixed in #343 AFAIA

**Comment by Roasbeef** [comment]
Closing.
